Source/Core/Core/MemWatcher.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+#include "Common/FileUtil.h"
+#include "Common/Thread.h"
+#include "Core/MemWatcher.h"
+#include "Core/HW/Memmap.h"
+
+namespace MW
+{
+
+// We don't want to kill the cpu, so sleep for this long after polling.
+static const int SLEEP_DURATION = 10; // ms
+
+static std::thread s_watcher_thread;
+static bool s_running = false;
+
+static int s_fd;
+static struct sockaddr_un s_addr;
+
+// Address -> last value.
+static std::map<u32, u32> s_values;
+
+static bool LoadAddresses()
+{
+	std::string locations_path = File::GetUserPath(F_MWCONFIG_IDX);
+	if (!File::Exists(locations_path))
+		return false;
+
+	std::ifstream ifs(locations_path.c_str(), std::ios::in);
```

Comment:
This should indicate what it's opening, `std::ios::in` isn't necessary for `ifstream` either, as it's an input stream, this is the default mode.